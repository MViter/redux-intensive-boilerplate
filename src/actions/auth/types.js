export default Object.freeze({
    LOGIN:          'LOGIN',
    LOGIN_SUCCESS:  'LOGIN_SUCCESS',
    LOGIN_FAIL:     'LOGIN_FAIL',

    SIGNUP:         'SIGNUP',
    SIGNUP_SUCCESS: 'SIGNUP_SUCCESS',
    SIGNUP_FAIL:    'SIGNUP_FAIL',

    LOGOUT:         'LOGOUT',
    LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
    LOGOUT_FAIL:    'LOGOUT_FAIL',

    LOGIN_GUEST: 'LOGIN_GUEST',
    LOGIN_GUEST_SUCCESS:  'LOGIN_GUEST_SUCCESS',
    LOGIN_GUEST_FAIL:     'LOGIN_GUEST_FAIL'
});