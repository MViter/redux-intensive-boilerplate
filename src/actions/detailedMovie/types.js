export default Object.freeze({
    FETCH_DETAILED_MOVIE:         'FETCH_DETAILED_MOVIE',
    FETCH_DETAILED_MOVIE_SUCCESS: 'FETCH_DETAILED_MOVIE_SUCCESS',
    FETCH_DETAILED_MOVIE_FAIL:    'FETCH_DETAILED_MOVIE_FAIL',
    CLEAR_DETAILED_MOVIE:         'CLEAR_DETAILED_MOVIE'
});
