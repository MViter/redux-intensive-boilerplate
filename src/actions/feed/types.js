// Types
export default Object.freeze({
    FETCH_NEW_MOVIES:         'FETCH_NEW_MOVIES',
    FETCH_NEW_MOVIES_SUCCESS: 'FETCH_NEW_MOVIES_SUCCESS',
    FETCH_NEW_MOVIES_FAIL:    'FETCH_NEW_MOVIES_FAIL',

    FETCH_POPULAR_MOVIES:         'FETCH_POPULAR_MOVIES',
    FETCH_POPULAR_MOVIES_SUCCESS: 'FETCH_POPULAR_MOVIES_SUCCESS',
    FETCH_POPULAR_MOVIES_FAIL:    'FETCH_POPULAR_MOVIES_FAIL',

    FETCH_EXPECTED_MOVIES:      'FETCH_EXPECTED_MOVIES',
    FETCH_EXPECTED_SUCCESS:     'FETCH_EXPECTED_MOVIES_SUCCESS',
    FETCH_EXPECTED_MOVIES_FAIL: 'FETCH_EXPECTED_MOVIES_FAIL',

    LIKE_MOVIE:         'LIKE_MOVIE',
    LIKE_MOVIE_SUCCESS: 'LIKE_MOVIE_SUCCESS',
    LIKE_MOVIE_FAIL:    'LIKE_MOVIE_FAIL'
});
