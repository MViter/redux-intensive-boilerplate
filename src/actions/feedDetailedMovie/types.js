export default Object.freeze({
    FETCH_DETAILED_MOVIE:         'FETCH_DETAILED_MOVIE',
    FETCH_DETAILED_MOVIE_SUCCESS: 'FETCH_DETAILED_MOVIE_SUCCESS',
    FETCH_DETAILED_MOVIE_FAIL:    'FETCH_DETAILED_MOVIE_FAIL'
});
