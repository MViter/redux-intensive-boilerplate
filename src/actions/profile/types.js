// Types
export default Object.freeze({
    FILL_USER_PROFILE:      'FILL_USER_PROFILE',
    CLEAR_USER_PROFILE:     'CLEAR_USER_PROFILE',
    UPDATE_PROFILE:         'UPDATE_PROFILE',
    UPDATE_PROFILE_SUCCESS: 'UPDATE_PROFILE_SUCCESS',
    UPDATE_PROFILE_FAIL:    'UPDATE_PROFILE_FAIL',
    UPDATE_AVATAR:          'UPDATE_AVATAR',
    UPDATE_AVATAR_SUCCESS:  'UPDATE_AVATAR_SUCCESS',
    UPDATE_AVATAR_FAIL:     'UPDATE_AVATAR_FAIL'
});
