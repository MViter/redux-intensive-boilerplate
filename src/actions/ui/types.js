export default Object.freeze({
    INITIALIZE:          'INITIALIZE',
    START_FETCHING_AUTH: 'START_FETCHING_AUTH',
    STOP_FETCHING_AUTH:  'STOP_FETCHING_AUTH'
});
