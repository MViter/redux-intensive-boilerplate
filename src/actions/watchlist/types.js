export default Object.freeze({
    FETCH_WATCHLIST:         'FETCH_WATCHLIST',
    FETCH_WATCHLIST_SUCCESS: 'FETCH_WATCHLIST_SUCCESS',
    FETCH_WATCHLIST_FAIL:    'FETCH_WATCHLIST_FAIL',

    ADD_TO_WATCHLIST:         'ADD_TO_WATCHLIST',
    ADD_TO_WATCHLIST_SUCCESS: 'ADD_TO_WATCHLIST_SUCCESS',
    ADD_TO_WATCHLIST_FAIL:    'ADD_TO_WATCHLIST_FAIL',

    DELETE_FROM_WATCHLIST:         'DELETE_FROM_WATCHLIST',
    DELETE_FROM_WATCHLIST_SUCCESS: 'DELETE_FROM_WATCHLIST_SUCCESS',
    DELETE_FROM_WATCHLIST_FAIL:    'DELETE_FROM_WATCHLIST_FAIL',

    CHECK_MOVIE_IN_WATCHLIST:         'CHECK_MOVIE_IN_WATCHLIST',
    CHECK_MOVIE_IN_WATCHLIST_SUCCESS: 'CHECK_MOVIE_IN_WATCHLIST_SUCCESS',
    CHECK_MOVIE_IN_WATCHLIST_FAIL:    'CHECK_MOVIE_IN_WATCHLIST_FAIL'
});
