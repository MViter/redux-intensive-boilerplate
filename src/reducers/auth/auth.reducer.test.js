// Core
import { Map } from 'immutable';

// Instruments
import reducer from './';
import types from 'actions/auth/types';

const initialState = Map({
    authenticated: false
});

describe('auth reducer:', () => {
    test(`should return initial state by default`, () => {
        expect(reducer(initialState, {})).toMatchSnapshot();
    });

    test(`should handle 'LOGIN_SUCCESS' and 'SIGNUP_SUCCESS' actions`, () => {
        expect(
            reducer(initialState, {
                type: types.LOGIN_SUCCESS
            })
        ).toMatchSnapshot();
        expect(
            reducer(initialState, {
                type: types.SIGNUP_SUCCESS
            })
        ).toMatchSnapshot();
    });

    test(`should handle 'LOGOUT_SUCCESS' action`, () => {
        expect(
            reducer(initialState, {
                type: types.LOGOUT_SUCCESS
            })
        ).toMatchSnapshot();
    });
});
