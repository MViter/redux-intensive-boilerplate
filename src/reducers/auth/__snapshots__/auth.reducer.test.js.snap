// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth reducer: should handle 'LOGIN_SUCCESS' and 'SIGNUP_SUCCESS' actions 1`] = `
Immutable.Map {
  "authenticated": true,
}
`;

exports[`auth reducer: should handle 'LOGIN_SUCCESS' and 'SIGNUP_SUCCESS' actions 2`] = `
Immutable.Map {
  "authenticated": true,
}
`;

exports[`auth reducer: should handle 'LOGOUT_SUCCESS' action 1`] = `
Immutable.Map {
  "authenticated": false,
}
`;

exports[`auth reducer: should return initial state by default 1`] = `
Immutable.Map {
  "authenticated": false,
}
`;
